biohansel/utils.py

```diff
@@ -170,30 +146,67 @@ def is_gzipped(p: str) -> bool:
+
+
+def collect_inputs(files,
+                   input_fasta_genome_name,
+                   input_directory,
+                   paired_reads,
+                   **kwargs) -> Tuple[List[Tuple[str, str]], List[Tuple[List[str], str]]]:
+    """Collect all input files for analysis
+
+    Sample names are derived from the base filename with no extensions.
+    Sequencing reads are paired if they share a common filename name without "_\d".
+    Filepaths for contigs and reads files are collected from an input directory if provided.
 
     Args:
         args: ArgumentParser.parse_args() output
-        scheme: Scheme name e.g. "heidelberg"
 
     Returns:
-        SubtypingParams with user-supplied values then scheme defaults then global defaults loaded
+        List of (contig filename, sample name)
+        List of ([reads filepaths], sample name)
     """
-    subtyping_params = get_scheme_params(scheme)
-    if subtyping_params is None:
-        subtyping_params = SubtypingParams()
-    if args is not None:
-        if args.low_cov_depth_freq:
-            subtyping_params.low_coverage_depth_freq = args.low_cov_depth_freq
-        if args.max_missing_tiles:
-            subtyping_params.max_perc_missing_tiles = args.max_missing_tiles
-        if args.min_ambiguous_tiles:
-            subtyping_params.min_ambiguous_tiles = args.min_ambiguous_tiles
-        if args.max_intermediate_tiles:
-            subtyping_params.max_perc_intermediate_tiles = args.max_intermediate_tiles
-        if args.low_cov_warning:
-            subtyping_params.min_coverage_warning = args.low_cov_warning
-
-    return subtyping_params
+    input_genomes = []
+    reads = []
+    if files:
+        fastas = [x for x in files if REGEX_FASTA.match(x)]
+        fastqs = [x for x in files if REGEX_FASTQ.match(x)]
+        if len(fastas) > 0:
+            for fasta_path in fastas:
+                fasta_path = os.path.abspath(fasta_path)
+                if os.path.exists(fasta_path):
+                    genome_name = genome_name_from_fasta_path(fasta_path)
+                    input_genomes.append((fasta_path, genome_name))
+                else:
+                    logging.error(f'Input fasta "{fasta_path}" does not exist!')
+        if len(fastqs) > 0:
+            grouped_fastqs = group_fastqs(fastqs)
+            logging.info(f'Grouped {len(fastqs)} fastqs into {len(grouped_fastqs)} groups')
+            reads += grouped_fastqs
+    if input_fasta_genome_name:
+        for fasta_path, genome_name in input_fasta_genome_name:
+            input_genomes.append((os.path.abspath(fasta_path), genome_name))
+    if input_directory:
+        logging.info(f'Searching dir "{input_directory}" for FASTA files')
+        input_genomes += collect_fasta_from_dir(input_directory)
+        logging.info(f'Searching dir "{input_directory}" for FASTQ files')
+        reads += collect_fastq_from_dir(input_directory)
+    if paired_reads:
+        for x in paired_reads:
```

Comment:
just a small thing, but perhaps a more descriptive variable name for x?

Reply:
Changed in 47fa20fae518aee051462ab5c9116422d5bd47db